dbt/include/clickhouse/macros/utils/utils.sql

```diff
@@ -63,11 +63,15 @@
 {% endmacro %}
 
 
-{% macro clickhouse__listagg(measure, delimiter_text, order_by_clause, limit_num) %}
-  {{ exceptions.raise_compiler_error(
-    'ClickHouse does not support the listagg function.  See the groupArray function instead')
-    }}
-{% endmacro %}
+{% macro clickhouse__listagg(measure, delimiter_text, order_by_clause, limit_num) -%}
+    {% set arr = "array_agg({})".format(measure) %}
+    {% set arr = "arraySort({}, {})".format(arr, order_by_clause) if order_by_clause else arr %}
+    {% if limit_num -%}
+      arrayStringConcat('{{delimiter_text}}', arraySlice({{ arr }}, 1, {{ limit_num }}))
```

Comment:
based on the [docs](https://clickhouse.com/docs/en/sql-reference/functions/splitting-merging-functions#arraystringconcat) it seems that you need to pass the column name/array first, and only then the delimiter.

Reply:
ah, thanks a lot!

Reply:
based on [docs](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros#listagg) singlequote also not needed. fixed that too